test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -301,6 +311,40 @@ func runFuncWithRecover(f pulumi.RunFunc) pulumi.RunFunc {
 	}
 }
 
+func shouldRetryError(err error) bool {
+	if auto.IsCompilationError(err) {
```

Comment:
Is this required? Do we happen to have an `err.Error()` that contains the matching string but should not be retried?

Reply:
I will remove it for readability. I feel like it should never happen